### PR TITLE
workflows/triage: add linux-only label

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This allows us to quickly identify bump PRs as updates to Linux formulae, which for the time being, should not be updated in `homebrew-core`; see https://github.com/Homebrew/homebrew-core/pull/64175#issuecomment-723311040

> I was not really expecting people to upgrade linux-only formulae here before completing the merging of both repositories. Especially as no bottles are built; and not reverse dependencies are tested. So we may introduce breakages in linuxbrew-core, which is far from ideal.

### CODE REVIEW

**Comment by SMillerDev** [comment]
This should also behave as a `do not merge`-label then for testbot.

**Comment by SMillerDev** [comment]
I don't think it's a good idea to add `do not merge` directly, I'd say it should be a new label that also blocks automerge. But ultimately dependencies not being tested in linuxbrew is the real problem that should be fixed.

**Review by MikeMcQuaid — DISMISSED** [review]
👍🏻 to `linux-only` label, agreed about not having a `do not merge` yet.

**Comment by iMichka** [comment]
I am ok for a `linux-only` label too.

> But ultimately dependencies not being tested in linuxbrew is the real problem that should be fixed.

They are tested, but in linuxbrew-core. Ideally one would first open a PR in linuxbrew-core, check if everything is fine, and then backport the change here.
Or not make any version bumps here for the time being, we are not in a hurry.

This situation will last until we have merged both repositories: we made huge progress but there are still a lot of things to finish.

**Comment by SMillerDev** [comment]
> They are tested, but in linuxbrew-core. Ideally one would first open a PR in linuxbrew-core, check if everything is fine, and then backport the change here.

IMHO that's an unworkable situation. I'd say we need to test it here, or not have it here at all. Otherwise it's a ton of work for everyone.

**Comment by MikeMcQuaid** [comment]
> This situation will last until we have merged both repositories: we made huge progress but there are still a lot of things to finish.

Do you have a (very) rough time estimate?

**Comment by iMichka** [comment]
The other strategy was to migrate all the linux changes to this repo in one huge PR, when we are ready. But this also means reviewing a PR that will contain more than 2000 formula changes, and this PR will be an endless moving target.

I prefer to add things here slowly, with the risk that some formulae are tested on Linux only with a delay. That does not change much to the current situation where homebrew-core formulae are only tested on Linux when we merge it to linuxbrew-core.

The idea is that the changes we make to the homebrew-core repo:
- are tested on mac and not break anything for mac users
- are tested on linux with delay, after the merge, as it has always been
- for some formulae are partially tested for linux in homebrew-core, on a best-effort basis

**Comment by iMichka** [comment]
> Do you have a (very) rough time estimate?

6 months? 1 year? It depends on how much help I get to migrate everything. I'll do a status update on https://github.com/Homebrew/brew/issues/7028 soon.

**Comment by SMillerDev** [comment]
> I prefer to add things here slowly, with the risk that some formulae are tested on Linux only with a delay. That does not change much to the current situation where homebrew-core formulae are only tested on Linux when we merge it to linuxbrew-core.

Is there really no way to properly test this in core on both platforms? Maybe a migration "per family" as we would do with things like Python is the way to go here? I'd love to help, but it's hard to keep track of the status for me.

**Comment by MikeMcQuaid** [comment]
> 6 months? 1 year? It depends on how much help I get to migrate everything. I'll do a status update on [Homebrew/brew#7028](https://github.com/Homebrew/brew/issues/7028) soon.

Thanks, that's helpful. Given that:

> * some formulae are partially tested for linux in homebrew-core, on a best-effort basis

I think the level of testing done here should be ramped up. It may not be feasible to do reverse dependency checking (although it's probably a good idea, IMO) but we could at least be creating and testing bottles without upload.

**Comment by iMichka** [comment]
> I think the level of testing done here should be ramped up. It may not be feasible to do reverse dependency checking (although it's probably a good idea, IMO) but we could at least be creating and testing bottles without upload.

Agreed.

I think reverse dependency checking only happens for formulae with a bottle on Linux, as we did not bottle everything until now.
The issue is that some formulae have different revisions on linux and on mac, so we can not just "fetch" bottles as-is from linux and use them for testing in the homebrew-core.

What I would like is to slowly introduce linux CI for a few "main" formuale in homebrew-core in the next months, to make sure these do not get broken by updates. I am thinking about things like gcc, pkg-config, openssl, unzip, perl, python ...
Of course these will not be able to use bottles for their builds, so we can not go too deep into the dependency tree.

>  👍🏻 to linux-only label, agreed about not having a do not merge yet.

Can this PR be changed to add just the first label.

After some thoughts: I am totally fine that things get bumped here, I was just surprised because I did not think it would happen. But it is just the logical thing to do. We will monitor breakages in linuxbrew-core after merges, like we do for all the other packages. Nothing really changed in the process.

**Comment by SMillerDev** [comment]
Just wondering, what's stopping us from already bottling the formula we have migrated here? That seems like the easiest thing. That way you "actually" migrate them rather than upstreaming the changes.

**Comment by iMichka** [comment]
We could, if it is acceptable to build bottles in this repo, step-by-step. I'll write some stuff about this in https://github.com/Homebrew/brew/issues/7028 then, as there are a few things to be discussed before we do that, and I do not want to hijack this PR for this discussion.

**Comment by SMillerDev** [comment]
I'd say we should make bottles here if we can. That's the end goal anyway.

**Comment by iMichka** [comment]
Cool. Let's discuss this in https://github.com/Homebrew/brew/issues/7028 for the practical details

For this PR: we would like to have only the `linux-only` label.

File: .github/workflows/triage.yml
Comment (SMillerDev) [inline .github/workflows/triage.yml:80]:
```suggestion
```

**Comment by iMichka** [comment]
Thanks all for the input!

**Pull request merged**
